File: spark-job/src/test/scala/com/dqs/SparkSessionTest.scala
package com.dqs

import org.scalatest.funsuite.AnyFunSuite
import org.apache.spark.sql.SparkSession

class SparkSessionTest extends AnyFunSuite {
  test("SparkSession can be initialized locally") {
    val spark = SparkSession.builder()
      .master("local[*]")
      .appName("DqsTest")
      .getOrCreate()

    assert(spark.version.startsWith("3.5"))
    spark.stop()
  }
}

File: spark-job/src/test/java/com/dqs/SparkSessionTest.java
package com.dqs;

import org.apache.spark.sql.SparkSession;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class SparkSessionTest {

    @Test
    void sparkSessionCanBeInitializedLocally() {
        SparkSession spark = SparkSession.builder()
                .master("local[*]")
                .appName("DqsTest")
                .getOrCreate();
        assertTrue(spark.version().startsWith("3.5"));
        spark.stop();
    }
}
